src/hooks/useTitle.ts

```diff
@@ -0,0 +1,14 @@
+import React, { useState, useEffect } from 'react'
+import ReactDOM from 'react-dom'
+
+export const useTitle = (initialTitle: string) => {
+    const [title, setTitle] = useState(initialTitle)
+
+    const updateTitle = () => {
+        const htmlTitle = document.querySelector('title')
+        htmlTitle!.innerText = title
```

Comment:
mam pytanko czy dla bezpieczenstwa nie lepiej użyć takiego zapisu: 
htmlTitle.innerText = `${title}`

Reply:
hm, bezpieczeństwa czego? :) ten title to useState, który będzie nadawany przez nas ręcznie w każdym widoku, np:

![Screen Shot 2023-05-02 at 21 11 16](https://user-images.githubusercontent.com/98549349/235762777-3d3d926b-266d-487b-8165-af0dec2c8e39.png)

Reply:
zapis z ${title} stosowalibyśmy np. w backtickach, a tutaj po prostu nadajemy innerTextowi wartość z title.

Reply:
ok masz racje, brałam pod uwagę sytuację gdyby zmienna title w przyszłości mogła przyjmować wartości, które są wprowadzane przez użytkownika lub pochodzą z niezaufanego źródła, i użycie template literals mogłoby zapobiec atakom typu XSS :)